### PR TITLE
Cleaned up script locations (/modules) and gulp configurations.

### PR DESCRIPTION
**Pull request by mandragorn**
moved code into flat modules directory for easy import of parts into other projects.
Simplified and streamlined gulp build
For gulp watch and gulp tests:watch to work gulp-jasmine needs to be updated with these changes: https://github.com/sindresorhus/gulp-jasmine/pull/43

### CODE REVIEW

**Comment by mandragorn** [comment]
bloody hell that was annoying.... look at the overall diff and once I address peer review comments I'll merge this into a single commit and push a new pull request.

File: README.md [hunk 1]
```diff
@@ -2,116 +2,82 @@
 A holistic approach to Backbone applications
 
 ## Build Commands:
```
Comment (kentmw) [inline README.md]: Let's chat about how we want to set up our readme. I think this is too much about the build.

File: README.md [hunk 2]
```diff
@@ -2,116 +2,82 @@
...
-npm run-script build
-npm run-script build:clean
+npm run build
 ```
 or
 ```
-node_modules/gulp/bin/gulp.js default
-node_modules/gulp/bin/gulp.js default:clean
-
-node_modules/gulp/bin/gulp.js build
```
Comment (kentmw) [inline README.md]: I think any references to gulp should be 'gulp' not 'node_modules/...gulp'

File: modules/torso.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    },
+    events: events,
+    Mixins: {
+      collectionLoading: collectionLoadingMixin,
+      collectionRegistration: collectionRegistrationMixin,
+      polling: pollingMixin,
+      validation: validation.mixin,
+      viewHierarchy: viewHierarchyMixin
+    },
+    Models: {
```
Comment (kentmw) [inline modules/torso.js]: this matches what is generated when requiring all the modules globally, correct?
Not a big fan of Torso.Models.Model, but we can think about that later.

File: test/spec/import/commonJsImports.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
+var _ = require('underscore'),
+    commonJsImportTest = function(moduleToImport, expectedModules) {
```
Comment (kentmw) [inline test/spec/import/commonJsImports.js]: define the function variable name before _ and then define the function without a tab

File: test/spec/import/globalImports.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
+describe('Global Module imports', function() {
+  var window, globalIndex,
+      globals = [
+        ['Torso'],
```
Comment (kentmw) [inline test/spec/import/globalImports.js]: Add a comment about what this set up does

File: test/spec/import/commonJsImports.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      expectedModules.push(moduleToImport);
+      describe('CommonJS Module import of "' + moduleToImport + '"', function() {
+        var windowRequire, module, actualModule, moduleIndex;
+
+        beforeAll(function(done) {
```
Comment (kentmw) [inline test/spec/import/commonJsImports.js]: give an example of what this does in comments

File: modules/torso.js (hunk 3 above)
Comment (mandragorn) [inline modules/torso.js]: Yea but before it was:
Torso.Model
Torso.Models.Nested
which seemed inconsistent to me.  At least this is consistent for all models.

**Pull request merged**